packages/edit-widgets/src/store/actions.js

```diff
@@ -108,17 +108,7 @@ export function* saveWidgetArea( widgetAreaId ) {
 		( { sidebar } ) => sidebar === widgetAreaId
 	);
 
-	// Remove all duplicate reference widget instances
-	const usedReferenceWidgets = [];
-	const widgetsBlocks = post.blocks.filter( ( { attributes: { id } } ) => {
-		if ( id ) {
-			if ( usedReferenceWidgets.includes( id ) ) {
-				return false;
-			}
-			usedReferenceWidgets.push( id );
-		}
-		return true;
-	} );
+	const widgetsBlocks = post.blocks;
```

Comment:
```js
const widgetsBlocks = post.blocks.filter( ( { clientId } ) => {
		if ( clientId ) {
			if ( usedReferenceWidgets.includes( clientId ) ) {
				return false;
			}
			usedReferenceWidgets.push( clientId );
		}
		return true;
	} );
```

Could we use `clientId` here as the identifier?

Reply:
I guess all the blocks have different `clientId` by default.  I don't think it will make any difference rather than just removing the filter itself. As it is blocking the blocks with a similar `id` to be saved. That's why I decided to remove it. 😅 

Thank you for taking a look! 😄

Reply:
Yeh you're absolutely right. Needed to fully understand the problem.